packages/@aws-cdk/lambda-layer-awscli/lib/awscli-layer.ts

```diff
@@ -8,8 +10,18 @@ import { Construct } from 'constructs';
 export class AwsCliLayer extends lambda.LayerVersion {
   constructor(scope: Construct, id: string) {
     super(scope, id, {
-      code: lambda.Code.fromAsset(path.join(__dirname, 'layer.zip')),
+      code: lambda.Code.fromAsset(path.join(__dirname, 'layer.zip'), {
+        // we hash the Dockerfile (it contains the tools versions) because hashing the zip is non-deterministic
+        assetHash: hashFile(path.join(__dirname, '..', 'layer', 'Dockerfile')),
+      }),
       description: '/opt/awscli/aws',
     });
   }
 }
+
+function hashFile(fileName: string) {
+  return crypto
+    .createHash('sha256')
+    .update(fs.readFileSync(fileName))
+    .digest('hex');
+}
```

Comment:
Should consolidate the function duplicate ?

Reply:
Maybe, but this will need to go in core and I didn't want to bother with it right now.